modal/app.py

```diff
@@ -696,6 +696,20 @@ def wrapped(
                 # typically for @function-wrapped @web_endpoint, @asgi_app, or @batched
                 f.wrapped = True
 
+                # but we don't support @app.function wrapping a method.
+                if is_method_fn(f.raw_f.__qualname__):
+                    raise InvalidError(
+                        "The `@app.function` decorator cannot be used on class methods. "
+                        "Please use `@app.cls` with `@modal.method` or `@modal.web_endpoint` instead. Example: "
```

Comment:
Is the suggestion here overly restrictive? There are other partial function decorators that could work.

Reply:
Yes it possibly is. I don't want to list all the alternative decorators though.